pkg/gce-cloud-provider/compute/cloud-disk.go

```diff
@@ -92,6 +92,17 @@ func (d *CloudDisk) GetKind() string {
 	}
 }
 
+func (d *CloudDisk) GetStatus() string {
+	switch d.Type() {
+	case Zonal:
+		return d.ZonalDisk.Status
+	case Regional:
+		return d.RegionalDisk.Status
+	default:
+		return ""
```

Comment:
nit: this will end up with an obscure error message in isDiskReady. Maybe something more informative could be returned here?